### PR TITLE
rule unique tags - <title> and <main>

### PR DESCRIPTION
**Pull request by DragorWW**
Add rule (unique-tags) for check unique tags <title> and <main> on page
add test for rule  (unique-tags)

### CODE REVIEW

File: src/rules/unique-tags.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        uniqueTags.forEach(function(item) {
+            map[item] = 0;
+        });
+
+        function onTagStart(event){
+            var tagName = event.tagName.toLowerCase();
+            if(typeof map[tagName] !== 'undefined'){
+
+                if (++map[tagName] > 1) {
+                    reporter.error('<' + tagName +'> must be one page', event.line, event.col, self, event.raw);
```
Comment (waldyrious) [inline src/rules/unique-tags.js]: I suggest 'there should be only one <' + tagName +'> per page

File: src/rules/unique-tags.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2015, DragorWW <DragorWW@gmail.com>
+ * The idea of rule is taken of philipwalton/html-inspector
+ * MIT Licensed
+ */
+HTMLHint.addRule({
+    id: 'unique-tags',
+    description: 'unique tags <title> or <main>, must be one on page!',
```
Comment (waldyrious) [inline src/rules/unique-tags.js]: perhaps "there can only be one <title> or <main> tag per page"?

Comment (DragorWW) [inline src/rules/unique-tags.js]: change message by your first comment

**Comment by DragorWW** [comment]
My English is poor, renamed error message and description of rule

File: src/rules/unique-tags.js (hunk 1 above)
Comment (DragorWW) [inline src/rules/unique-tags.js]: changed

**Comment by yaniswang** [comment]
This rule may little match, can you show your case?
More rule will reduce performance.

**Comment by DragorWW** [comment]
At my company, we decided to check using HTML Hint ready html and integrated page.
Checks must fulfill two functions:
1, additionaly protect from fools (2 title for example)
1 check guildline

does not necessarily include all of the default rules

**Comment by yaniswang** [comment]
I mean this rule may matched < 0.1%, this case will appared nothing much.

**Comment by DragorWW** [comment]
It may then accept my pr https://github.com/yaniswang/HTMLHint/pull/82, so that you can add custom rules?

**Pull request closed without merging**